packages/typescript-estree/src/node-utils.ts

```diff
@@ -471,15 +475,11 @@ export function isChildUnwrappableOptionalChain(
     | ts.NonNullExpression,
   child: TSESTree.Node,
 ): boolean {
-  if (
+  return (
     isChainExpression(child) &&
     // (x?.y).z is semantically different, and as such .z is no longer optional
     node.expression.kind !== ts.SyntaxKind.ParenthesizedExpression
-  ) {
-    return true;
-  }
-
-  return false;
+  );
```

Comment:
coverage changed to this, as this change simply reduces cyclomatic complexity of this function